larq_zoo/literature/meliusnet.py

```diff
@@ -0,0 +1,231 @@
+from typing import Optional, Sequence, Union
+
+import larq as lq
+import tensorflow as tf
+from zookeeper import Field, factory
+
+from core import utils
+from core.model_factory import ModelFactory
+
+
+@factory
+class MeliusNetFactory(ModelFactory):
+    weights: Optional[str] = Field(None)
+
+    # overall architecture configuration
+    num_blocks: Sequence[int]
+    transition_features: Sequence[int]
+    name: str
+
+    # Some default layer arguments.
+    batch_norm_momentum: float = Field(0.9)
+    kernel_initializer: Optional[Union[str, tf.keras.initializers.Initializer]] = Field(
+        "glorot_normal"
+    )
+    input_quantizer = Field(lq.quantizers.SteSign(1.3))
+    kernel_quantizer = Field(lq.quantizers.SteSign(1.3))
+    kernel_constraint = Field("weight_clip")
```

Comment:
I am pretty sure you should also change the `kernel_constraint`:
```suggestion
    input_quantizer = Field(lambda: lq.quantizers.SteSign(1.3))
    kernel_quantizer = Field(lambda: lq.quantizers.SteSign(1.3))
    kernel_constraint = Field(lambda: lq.constraints.WeightClip(1.3))
```